src/tests/ftest/soak/utils.py

```diff
@@ -1196,6 +1195,71 @@ def create_fio_cmdline(self, job_spec, pool):
     return commands
 
 
+def create_app_cmdline(self, job_spec, pool, ppn, nodesperjob):
+    """Create the srun cmdline to run app.
+
+    This method will use a cmdline specified in the yaml file to
+    execute a local binary until the rpms are available
+    Args:
+        self (obj):       soak obj
+        job_spec (str):   job in yaml to run
+        pool (obj):       TestPool obj
+        ppn(int):         number of tasks to run on each node
+        nodesperjob(int): number of nodes per job
+    Returns:
+        cmd(list): list of cmdlines
+
+    """
+    commands = []
+    sbatch_cmds = []
+    app_params = os.path.join(os.sep, "run", job_spec, "*")
+    app_cmd = self.params.get("cmdline", app_params, default=None)
+    mpi_module = self.params.get("module", app_params, self.mpi_module)
+    posix = self.params.get("posix", app_params, default=False)
+    if app_cmd is None:
+        self.log.info(
+            "<<{} command line not specified in yaml; job will not be run>>".format(job_spec))
+        return commands
+    oclass_list = self.params.get("oclass", app_params)
+    for oclass in oclass_list:
+        add_containers(self, pool, oclass)
+        sbatch_cmds = ["module purge", "module load {}".format(self.mpi_module)]
+        log_name = "{}_{}_{}_{}_{}".format(
+            job_spec, oclass, nodesperjob * ppn, nodesperjob, ppn)
+        # include dfuse cmdlines
+        if posix:
+            dfuse, dfuse_start_cmdlist = start_dfuse(
+                self, pool, self.container[-1], name=log_name, job_spec=job_spec)
+            sbatch_cmds.extend(dfuse_start_cmdlist)
+        # allow apps that use an mpi other than default (self.mpi_module)
+        if mpi_module != self.mpi_module:
+            sbatch_cmds.append("module load {}".format(mpi_module))
+        mpirun_cmd = Mpirun(app_cmd, False, mpi_module)
+        if "mpich" in mpi_module:
+            # Pass pool and container information to the commands
+            env = EnvironmentVariables()
+            env["DAOS_UNS_PREFIX"] = "daos://{}/{}/".format(pool.uuid, self.container[-1].uuid)
```

Comment:
FYI, there is a function for this, though it doesn't necessarily shorten the code required
https://github.com/daos-stack/daos/blob/d1c028d7db27427674b99189b2a683b9f7149c2a/src/tests/ftest/util/duns_utils.py#L8

```
import duns_utils
duns_utils.format_path(pool.uuid, self.container[-1].uuid)
```